### PR TITLE
Joomla 4 & PHP8.0 Compatability

### PR DESCRIPTION
**Pull request by vinayak-patil**
Have tested all the APIs for Joomla 4 and PHP 8 compatibility

Version Used -
Joomla 4.2.8  - Joomla 3.10.11
PHP 8.0 - PHP 7.3

### CODE REVIEW

File: categories/categories/categories.php [hunk 1]
```diff
@@ -83,6 +89,20 @@ public function getCategoriesList()
...
+		{			
+			$obj->success = true;
+			$obj->data = $rows;
+
+			return $obj;
+		}
+		else
+		{
+			$obj->success = false;
+			$obj->message = 'System does not have categories';
```
Comment (ankush-maherwal) [inline categories/categories/categories.php]: @vinayak-patil Use language constant for text "System does not have categories"

File: articles/articles/helper/simpleschema.php [hunk 2]
```diff
@@ -60,13 +62,13 @@ public function mapPost($blog, $strip_tags='', $text_length=0, $skip=array())
 		else
 		{
 			//get media from custom component
-			$media_obj = new TZ_PortfolioModelMedia();
-			$med = $media_obj->getMedia($blog->id);
+			// $media_obj = new TZ_PortfolioModelMedia();
```
Comment (ankush-maherwal) [inline articles/articles/helper/simpleschema.php]: @vinayak-patil Why we have commented this code? What was it used for?

File: categories/categories/categories.php (hunk 1 above)
Comment (vinayak-patil) [reply categories/categories/categories.php]: @ankush-maherwal Thank you for your guidance. Will revise and update you.

File: articles/articles/helper/simpleschema.php (hunk 2 above)
Comment (vinayak-patil) [reply articles/articles/helper/simpleschema.php]: @ankush-maherwal Hi, This code is used to format the response, to get media(image) from 3rd party extension TZ_Portfolio.

So if we are accessing articles using latest resource (option=com_api&app=articles&resource=latest) and if TZ_Portfolio extension is not exist then we are getting error '0 Class "TZ_PortfolioModelMedia" not found'

Comment (vinayak-patil) [reply articles/articles/helper/simpleschema.php]: Hi @ankush-maherwal , As this is related to TZ_Portfolio extension and I don't know the use and working of this extension so I am restoring this code.

**Pull request merged**